### PR TITLE
[MRG+1] Minor cleanup of hist().

### PR DESCRIPTION
**Pull request by anntzer**
Mostly style, use `clip`, get rid of a redundant conditional.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -6273,28 +6273,23 @@ def _normalize_input(inp, ename='input'):
...
-                width = dr*totwidth/nx
+                width = dr * totwidth / nx
                 dw = width
-
-                if nx > 1:
-                    boffset = -0.5*dr*totwidth*(1.0-1.0/nx)
-                else:
-                    boffset = 0.0
-                stacked = False
+                boffset = -0.5 * dr * totwidth * (1 - 1 / nx)
```
Comment (madphysicist) [inline lib/matplotlib/axes/_axes.py]: Doesn't this introduce the possibility of divide-by-zero?

Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: l.6284 also divides by nx, so if nx is 0, we already to a divide-by-zero.

Comment (madphysicist) [inline lib/matplotlib/axes/_axes.py]: Good point.

Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: if `nx` is zero, then there is no data so other things are confused as well.

**Comment by NelleV** [comment]
LGTM
Thanks @anntzer !

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (afvincent) [inline lib/matplotlib/axes/_axes.py]: It seems to me that due to l. 6176-6181 (and above), the case `nx == 0` should never occur: the case where the input is empty is preprocessed to have a length of 1.
```python
# Massage 'x' for processing.
if input_empty:
    x = np.array([[]])
else:
    x = _normalize_input(x, 'x')
nx = len(x)  # number of datasets
```

**Comment by tacaswell** [comment]
As much as touching `hist` makes me nervous....

**Pull request merged**